src/Raven.Server/Documents/Indexes/Workers/MapDocuments.cs

```diff
@@ -79,6 +79,7 @@ public MapDocuments(Index index, DocumentsStorage documentsStorage, IndexStorage
                             {
                                 while (true)
                                 {
+                                    _index.CreatedFieldsCount = 0;
```

Comment:
We have other works also. This needs to be reset per document, correct? We have Cleanup worker (do not think it is needed there), but for sure in HandleItems (HandleReferences) one? What about Map-Reduce?